deeds-tenant-service/src/main/java/io/meeds/tenant/metamask/web/MetamaskLoginExtension.java

```diff
@@ -34,6 +36,16 @@
 @Service
 public class MetamaskLoginExtension extends BaseMetamaskExtension {
 
+  private static final String            METAMASK_LOGIN = "metamaskLogin";
+
+  protected            ExoFeatureService exoFeatureService;
+
+  @Autowired
+  public MetamaskLoginExtension(ExoFeatureService exoFeatureService) {
+    super();
+    this.exoFeatureService = exoFeatureService;
+  }
```

Comment:
```suggestion
  @Autowired
  protected            ExoFeatureService exoFeatureService;

```

Reply:
No need to inject through Constructor

Reply:
In fact, Sonar shows a warning with @Autowired on a field. we have the same on Kudos for example  : 
https://sonarcloud.io/project/issues?impactSoftwareQualities=RELIABILITY&resolved=false&id=org.exoplatform.addons.kudos&open=AY0dFb1SnUftKtTnJyga

> Generally speaking, field injection is discouraged. It allows the creation of objects in an invalid state and makes testing more difficult. The dependencies are not explicit when instantiating a class that uses field injection.

That's why I put it in constructor

Reply:
Yes, in fact I 'm aware of this minor recommendation (`SonarLint` reminds me the same in the IDE), but it would be better to provide an easy way to distinguish for now the injection using Spring beans and using Kernel IoC. Besides, the code readability is enhanced when no constructor is used (I think of switching process in which case, developers has to make sure to use the `@Autowired` on fields which helps to know which bean is managed through beans). For Maintainability purpose as well, in contrary to what Sonar says, when we add a new dependent service, we need as well to remember to inject it in constructor parameters, thus more code to have the same outcome.

Reply:
I understand that you prefer field injection as it is easier to develop, to read, and generate less code. 

But it seems to have some disadvantages : 
- https://medium.com/@detoxicdev/field-injection-v-s-constructor-injection-dd9db2d85b7b
- https://www.baeldung.com/java-spring-field-injection-cons
- https://www.linkedin.com/pulse/you-should-stop-using-spring-autowired-felix-coutinho/

As some arguments are related to clean code architecture, to testability, and to design problems, would it be preferable to use constructor injection instead of field injection, and prefer clean and testable code to readability ?

Reply:
This seems very controversial as you can see in all discussions about the subject. For my part, as another argument in addition to those above, it's better to avoid this for better maintainability and code clean because we intend to add some initialization code (as you can see it [here by example](https://github.com/Meeds-io/gatein-portal/blob/34c195cb0b83b10fb78e878650fde37106041598/component/identity/src/main/java/org/exoplatform/services/organization/idm/PicketLinkIDMOrganizationServiceImpl.java#L75-L105)) when we have a constructor and thus by design, I would prefer as well avoid code complexity.
In concrete, the Cons against the Field injection doesn't seems to be sufficiently having advantages in our context.

Reply:
Ok, So I change for filed injections. 
Should we deactivate sonar rule ?

Reply:
> Ok, So I change for filed injections. Should we deactivate sonar rule ?

Yes, I think it's better

Reply:
ok, done for meeds-io org and for exo org.

Thanks